src/formats/__tests__/json-schema.spec.ts

```diff
@@ -0,0 +1,234 @@
+import {
+  isJSONSchema,
+  isJSONSchema2019_09,
+  isJSONSchemaDraft4,
+  isJSONSchemaDraft6,
+  isJSONSchemaDraft7,
+  isJSONSchemaLoose,
+} from '../json-schema';
+
+describe('JSON Schema format', () => {
+  describe('JSON Schema strict', () => {
+    it.each([
+      'http://json-schema.org/schema#',
+      'https://json-schema.org/schema#',
+      'http://json-schema.org/hyper-schema#',
+      'http://json-schema.org/draft-04/schema#',
+      'http://json-schema.org/draft-04/hyper-schema#',
+      'http://json-schema.org/draft-06/schema#',
+      'http://json-schema.org/draft-06/hyper-schema#',
+    ])('recognizes %s schema correctly', $schema => {
+      expect(isJSONSchema({ $schema })).toBe(true);
+    });
+
+    it('does not recognize invalid document', () => {
+      expect(isJSONSchema({ $schema: '2.0' })).toBe(false);
+      expect(isJSONSchema({ $schema: 'json-schema' })).toBe(false);
+      expect(isJSONSchema({ $schema: 2 })).toBe(false);
+      expect(isJSONSchema({ swagger: null })).toBe(false);
+      expect(isJSONSchema({ allOf: [] })).toBe(false);
+      expect(isJSONSchema({ type: 'string' })).toBe(false);
+      expect(isJSONSchema({})).toBe(false);
+      expect(isJSONSchema(null)).toBe(false);
+    });
+  });
+
+  describe('JSON Schema loose', () => {
+    describe('by $schema', () => {
+      it.each([
+        'http://json-schema.org/schema#',
+        'https://json-schema.org/schema#',
+        'http://json-schema.org/hyper-schema#',
+        'http://json-schema.org/draft-04/schema#',
+        'http://json-schema.org/draft-04/hyper-schema#',
+        'http://json-schema.org/draft-06/schema#',
+        'http://json-schema.org/draft-06/hyper-schema#',
+      ])('recognizes %s schema correctly', $schema => {
+        expect(isJSONSchemaLoose({ $schema })).toBe(true);
+      });
+    });
+
+    describe('by type', () => {
+      it.each(['any', 'array', 'boolean', 'integer', 'null', 'number', 'object', 'string'])(
```

Comment:
```suggestion
      it.each(['array', 'boolean', 'integer', 'null', 'number', 'object', 'string'])(
```

actually I changed my mind let's not talk about any its gonna confuse things down the line

Reply:
Sure, makes sense.